app/views/finance/invoices/_invoices.html.haml

```diff
@@ -26,6 +26,10 @@
         %td= link_to Delivery.model_name.human, [invoice.supplier,invoice.delivery] if invoice.delivery
         %td= link_to format_date(invoice.order.ends), new_finance_order_path(order_id: invoice.order_id) if invoice.order
         %td= truncate(invoice.note)
-        %td= link_to t('ui.edit'), edit_finance_invoice_path(invoice), class: 'btn btn-mini'
-        %td= link_to t('ui.delete'), finance_invoice_path(invoice), :data => {:confirm => t('.confirm_delete')}, :method => :delete,
-          class: 'btn btn-danger btn-mini'
+        %td
+          - if current_user.role_finance? or (current_user.role_invoices? and not invoice.paid_on)
```

Comment:
@Cor-Jan not sure if you can make anything out of this discussion, but what do you think of this? The idea is to allow not just financial people to add invoices, and only allow financial people to set invoices to being paid. What would be better?
1. All people who have the invoices permission can edit all unpaid invoices.
2. All people who have the invoices permission can create new invoices and edit their own unpaid ones.